### ISSUE TITLE
Issues installing multi-stage

### ISSUE THREAD

**Issue by stefanpenner** [npm/npm#8643]
This may be something on my end, but I figure I would post it.

It appears, installing via the git repo via `npm i -g ./` or `npm i -g "npm/npm#multi-stage"` fails to install `are-we-there-yet`

I have tried various node versions and npm versions, but seem to be without any luck.

node versions attempted (and there corresponding npm versions)

* 1.8.1
* 2.0.0
* 2.0.2
* 2,2,1

```
  % npm -v 
module.js:334
    throw err;
          ^
Error: Cannot find module 'are-we-there-yet'
    at Function.Module._resolveFilename (module.js:332:15)
    at Function.Module._load (module.js:282:25)
    at Module.require (module.js:361:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/Users/stefanpenner/.nave/installed/2.2.1/lib/node_modules/npm/node_modules/npmlog/log.js:2:16)
    at Module._compile (module.js:426:26)
    at Object.Module._extensions..js (module.js:444:10)
    at Module.load (module.js:351:32)
    at Function.Module._load (module.js:306:12)
    at Module.require (module.js:361:17)
```

**Comment by stefanpenner** [comment]
after manually installing `are-we-there-yet` it appears `gauge` is the next missing.

**Comment by stefanpenner** [comment]
having @othiym23 next to be is handy, apparently the following incantation is required.

```
node . install -g ./
```

TL;DR npm 2.x is unable to package npm 3.x

**Comment by stefanpenner** [comment]
All appeared to be going well, unfortunately I have run into a 'Maximum call stack size exceeded' issue.

````
npm ERR! Darwin 14.3.0
npm ERR! argv "/Users/stefanpenner/.nave/installed/1.8.1/bin/iojs" "/Users/stefanpenner/src/npm" "install" "-g" "./"
npm ERR! node v1.8.1
npm ERR! npm  v3.0.0-pre.13

npm ERR! Maximum call stack size exceeded
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

**Comment by iarna** [comment]
This is, unsurprisingly, not reproducible for me.

My guess is that there's something in your global modules that's triggering this bug. If you run:

`/Users/stefanpenner/src/npm ls -g`

Do you get the same error?

**Comment by stefanpenner** [comment]
> iojs /Users/stefanpenner/src/npm ls -g

seems to hang at 100% :(

**Comment by stefanpenner** [comment]
I have pruned my `/usr/local/node_modules` and it no longer pins the CPU, instead completes nicely as expected.

Lets assume I had something strange in `/usr/local/node_modules`.

**Comment by stefanpenner** [comment]
the secondary issue seems to be resolved. The primary, likely requires some sort of documentation so users are aware of the specific installation rules.

**Comment by iarna** [comment]
@stefanpenner: I'll add a `prepublish` that verifies that you're using a compatible version of npm and gives a sensible error if you aren't.

**Comment by stefanpenner** [comment]
> @stefanpenner: I'll add a prepublish that verifies that you're using a compatible version of npm and gives a sensible error if you aren't.

even better

**Comment by iarna** [comment]
That landed as https://github.com/npm/npm/commit/c1929bfe85497f5fefd8991fcc71c92a901991bc and looks like: 
![no-compat-version](https://cloud.githubusercontent.com/assets/983798/8299900/b9b09cca-194c-11e5-8fca-b293b3f8cabc.jpg)

**Comment by stefanpenner** [comment]
@iarna :+1:
